src/python/pants/backend/go/util_rules/third_party_pkg.py

```diff
@@ -3,28 +3,34 @@
 
 from __future__ import annotations
 
+import json
 import logging
-import os.path
+import os
+import posixpath
```

Comment:
`PurePath` from `pathlib` is probably better - it's what we normally use. And some users have a dream of Pants working with Windows, so `pathlib` makes that less painful.

Reply:
Go import paths always use `/` as the separator even on Windows. This code should not be using platform-specific path separators which could differ from `/` given that fact.

Reply:
Got it. Probably worth a comment at the call site of `posixpath`.

Reply:
Seems a bit more complicated than I first thought. We use `.relpath` (which doesn't have a `pathlib` equivalent I believe) to calculate the package's path under a module's root path. That should use `os.path.relpath` instead of the `posixpath.join` that it uses now. But then when appending to the module's import path, it should uses`/` as the separator so the code should split the package path into pieces and concatenate using `/`. So the code using `posixpath` needs a rewrite then.

Reply:
Rewritten as outlined.

Reply:
The `pathlib` method is `PurePath.relative_to()`: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to